src/dtclient/kubernetes_settings_test.go

```diff
@@ -0,0 +1,389 @@
+		dtc, err := NewClient(dynatraceServer.URL, apiToken, paasToken, skipCert)
+		require.NoError(t, err)
+		require.NotNil(t, dtc)
+
+		// act
+		actual, err := dtc.(*dynatraceClient).CreateKubernetesSetting(testName, "", testScope)
+
+		// assert
+		assert.Error(t, err)
+		assert.Len(t, actual, 0)
+	})
+
+	t.Run(`don't create settings for the given monitored entity id because of api error`, func(t *testing.T) {
+		// arrange
+		dynatraceServer := httptest.NewServer(dynatraceServerSettingsHandler(1, testObjectID, true))
+		defer dynatraceServer.Close()
+
+		skipCert := SkipCertificateValidation(true)
+		dtc, err := NewClient(dynatraceServer.URL, apiToken, paasToken, skipCert)
+		require.NoError(t, err)
+		require.NotNil(t, dtc)
+
+		// act
+		actual, err := dtc.(*dynatraceClient).CreateKubernetesSetting(testName, testUID, testScope)
+
+		// assert
+		assert.Error(t, err)
+		assert.Len(t, actual, 0)
+	})
+}
+
+func handleEntitiesRequest(request *http.Request, writer http.ResponseWriter, entities []MonitoredEntity) {
+	if request.Method == http.MethodGet {
+		if !strings.Contains(request.Form.Get("entitySelector"), "type(KUBERNETES_CLUSTER)") {
+			writer.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		meResponse := monitoredEntitiesResponse{
+			TotalCount: len(entities),
+			PageSize:   500,
+			Entities:   entities,
+		}
+
+		entitiesResponse, err := json.Marshal(meResponse)
+
+		if err != nil {
+			return
+		}
+
+		writer.WriteHeader(http.StatusOK)
+		writer.Write(entitiesResponse)
+	} else {
+		writeError(writer, http.StatusMethodNotAllowed)
+	}
+}
+
+func handleSettingsRequest(request *http.Request, writer http.ResponseWriter, totalCount int, objectId string) {
+	if request.Method == http.MethodGet {
+		if request.Form.Get("schemaIds") != "builtin:cloud.kubernetes" || request.Form.Get("scopes") == "" {
+			writer.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		settingsGetResponse, err := json.Marshal(GetSettingsResponse{TotalCount: totalCount})
+
+		if err != nil {
+			return
+		}
+
+		writer.WriteHeader(http.StatusOK)
+		writer.Write(settingsGetResponse)
+	} else if request.Method == http.MethodPost {
+		if request.Body == nil {
+			writer.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		body, err := ioutil.ReadAll(request.Body)
+
+		if err != nil {
+			return
+		}
+
+		var parsedBody []postKubernetesSettingsBody
+		err = json.Unmarshal(body, &parsedBody)
+
+		if err != nil {
+			return
+		}
+
+		var settingsPostResponse []postSettingsResponse
+		settingsPostResponse = append(settingsPostResponse, postSettingsResponse{
+			ObjectId: objectId,
+		})
+
+		settingsPostResponseBytes, err := json.Marshal(settingsPostResponse)
+
+		if err != nil {
+			return
+		}
+
+		writer.WriteHeader(http.StatusOK)
+		writer.Write(settingsPostResponseBytes)
+	} else {
+		writeError(writer, http.StatusMethodNotAllowed)
+	}
+
+}
+
+func dynatraceServerEntitiesHandler(entities []MonitoredEntity, isError bool) http.HandlerFunc {
```

Comment:
same as before
`mockDynatraceServerEntitiesHandler ` or something similar

Reply:
We tried to keep it similar to `dynatrace_client_test.go` which has `dynatraceServerHandler()`.
Nevertheless, I agree with your point and will change it - just a FYI for possible future refactorings.